src/test/groovy/life/qbic/utils/NanoporeParserSpec.groovy

```diff
@@ -0,0 +1,29 @@
+package life.qbic.utils
+
+import spock.lang.Specification
+
+import java.nio.file.Paths
+
+class NanoporeParserSpec extends Specification {
+
+    def exampleDirectoriesRoot = this.getClass().getResource("/dummyFileSystem/nanopore-instrument-output").getPath()
+
+    def "parsing a valid file structure creates a Map"() {
+        given:
+        def pathToDirectory = Paths.get(exampleDirectoriesRoot, "validates/QABCD001AB_E12A345a01_PAE12345")
+        print(pathToDirectory)
+        when:
+        def map = NanoporeParser.parseFileStructure(pathToDirectory)
+        then:
+        map instanceof Map
+    }
+
+    def "parsing an empty directory returns null"() {
+        given:
+        def pathToDirectory = Paths.get(exampleDirectoriesRoot, "fails/empty_directory")
+        when:
+        def map = NanoporeParser.parseFileStructure(pathToDirectory)
+        then:
+        map == null
+    }
```

Comment:
Parsing an empty directory should raise an exception!

Reply:
How to do that best? 
```groovy
assert ! name.isEmpty() && ! path.isEmpty() && ! fileType.isEmpty()

def convertedFile = [
    "name"     : name,
    "path"     : path,
    "file_type": fileType
]
```

Reply:
no, the parseFileStructure() method needs to throw an exception, the test needs to confirm that it is thrown if sth goes wrong.

Reply:
It does not so this can be resolved?